File: fixtures/python/type_inference/cross_file/service.py
"""Service layer exercising cross-file constructor + method calls."""

from models import User, load_user


def render_greeting(user: User) -> str:
    return user.greet()


def main() -> None:
    user = load_user("Alice")
    print(render_greeting(user))
    print(user.greet())

File: fixtures/php/type_inference/cross_file/Service.php
<?php

require_once 'User.php';

use App\Models\User;
use App\Models\loadUser;

function renderGreeting(User $user): string {
    return $user->greet();
}

$user = loadUser("Alice");
echo renderGreeting($user) . "\n";
echo $user->greet() . "\n";
